process/tags_v2.go

```diff
@@ -167,7 +167,7 @@ func iterateV2(buffer []byte, tagIndex int, cb func(i, total int, tag string) bo
 }
 
 func unsafeIterateV2(buffer []byte, tagIndex int, cb func(i, total int, tag []byte) bool) {
-	if len(buffer[1:]) < lenUint32 {
+	if len(buffer) <= 0 || len(buffer[1:]) < lenUint32 || tagIndex < 0 {
```

Comment:
> len(buffer) <= 0
minor nit, but len(buffer) will never be less than zero, just equal to zero

> len(buffer[1:]) < lenUint32

I think this case is better checked by checking the return value of `footerPosition` to make it clear what's happening